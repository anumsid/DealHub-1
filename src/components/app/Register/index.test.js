import React from 'react'
import { shallow } from 'enzyme'
import { stub } from 'sinon'
import toJson from 'enzyme-to-json'
import { SignUp } from './index'

describe('Register Component', () => {
  let wrapper
  let onClick
  beforeEach(() => {
    onClick = stub()
    const props = {
      errorMessage: null,
      onClick: onClick
    }
    wrapper = shallow(<SignUp {...props} />)
  })

  it('Matches snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('Handles click on form submit', () => {
    wrapper.find('#name').simulate('change', {target: {value: 'James'}})
    wrapper.find('#age').simulate('change', {target: {value: '32'}})
    wrapper.find('#gender').simulate('change', {target: {innerText: 'Male'}})
    wrapper.find('#city').simulate('change', {target: {value: 'New York'}})
    wrapper.find('#email').simulate('change', {target: {value: 'test@test.com'}})
    wrapper.find('#password').simulate('change', {target: {value: 'password'}})
    wrapper.find('FormButton').simulate('click')
    expect(onClick.calledOnce).toBe(true)
  })
})
