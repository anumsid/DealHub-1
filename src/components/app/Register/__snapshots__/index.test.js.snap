// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register Component Matches snapshot 1`] = `
<Form
  as="form"
  className="mx-auto"
  style={
    Object {
      "marginTop": 100,
      "width": "40rem",
    }
  }
>
  <FormInput
    as={[Function]}
    control={[Function]}
    fluid={true}
    id="name"
    label="Name"
    onChange={[Function]}
    placeholder="First, last"
  />
  <FormInput
    as={[Function]}
    control={[Function]}
    fluid={true}
    id="age"
    label="Age"
    onChange={[Function]}
    placeholder="Age"
  />
  <FormSelect
    as={[Function]}
    control={[Function]}
    fluid={true}
    id="gender"
    label="Gender"
    onChange={[Function]}
    options={
      Array [
        Object {
          "key": "m",
          "text": "Male",
          "value": "male",
        },
        Object {
          "key": "f",
          "text": "Female",
          "value": "female",
        },
      ]
    }
    placeholder="Gender"
  />
  <FormInput
    as={[Function]}
    control={[Function]}
    fluid={true}
    id="city"
    label="City"
    onChange={[Function]}
    placeholder="City"
  />
  <FormInput
    as={[Function]}
    control={[Function]}
    fluid={true}
    id="email"
    label="Email"
    onChange={[Function]}
    placeholder="Email"
    type="email"
  />
  <FormInput
    as={[Function]}
    control={[Function]}
    fluid={true}
    id="password"
    label="Password"
    onChange={[Function]}
    placeholder="Password"
    type="password"
  />
  <FormButton
    as={[Function]}
    control={[Function]}
    onClick={[Function]}
  >
    Signup
  </FormButton>
</Form>
`;
